compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -498,6 +501,80 @@ BType resolvePatternTypeFromMatchExpr(BLangErrorMatchPattern errorMatchPattern,
         return symTable.noType;
     }
 
+    public boolean isExpressionInUnaryValid(BLangExpression expr) {
+        if (expr.getKind() == NodeKind.GROUP_EXPR) {
+            // To resolve ex: -(45) kind of scenarios
+            return ((BLangGroupExpr) expr).expression.getKind() == NodeKind.NUMERIC_LITERAL;
+        } else {
+            return expr.getKind() == NodeKind.NUMERIC_LITERAL;
+        }
```

Comment:
Needs to be checked recursively. See https://github.com/ballerina-platform/ballerina-lang/pull/34622#issuecomment-1166823070.

Reply:
created  #36740 to track